### PR TITLE
Balances for Bitcoin chain

### PR DESCRIPTION
**Pull request by longcnttbkhn**
Hi everyone, I plan to build balances table for bitcoin chain, if you are interested you can join me.

### CODE REVIEW

**Comment by Longcnttbkhn** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jeff-dude** [comment]
@Hosuke can you give this the initial review?

File: models/transfers/bitcoin/satoshi/transfers_bitcoin_satoshi.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{{ config(materialized='view', alias = alias('satoshi'),
+        tags = ['dunesql'],
+        post_hook='{{ expose_spells(\'["bitcoin"]\',
+                                    "sector",
+                                    "transfers",
+                                    \'["longnhbkhn"]\') }}') }}
+with 
+    input_transfers as (
+        select
+            CAST('input' AS VARCHAR(5)) || CAST('-' AS VARCHAR(1)) || CAST(tx_id AS VARCHAR(100)) || CAST('-' AS VARCHAR(1)) || CAST(index AS VARCHAR(100)) || CAST('-' AS VARCHAR(1)) || CAST(address AS VARCHAR(100)) as unique_transfer_id,
```
Comment (Hosuke) [inline models/transfers/bitcoin/satoshi/transfers_bitcoin_satoshi.sql:10]: I will add a new field `type` and refactor the unique_key.

**Comment by Longcnttbkhn** [comment]
@Hosuke I think adding `block_height` to `unique_key` is not a correct solution in this case, the cause of this duplication is due to the reorg phenomenon, you can see details here https://www. alchemy.com/overviews/what-is-a-reorg
If execute the following query:
```
select * from bitcoin.inputs where tx_id = 0xec4104b77e2efadea0e479fb0aab2d607bd45b5708613c23cbc4af78ab0ef6bd
and index = 0 and address = 'bc1qxh2k85p64y2xpftgcjf2cam9khqnm3897tmtyp'
```
We will receive 2 records located in 2 consecutive blocks 804900 and 804901) however if we check on https://live.blockcypher.com/btc/tx/ec4104b77e2efadea0e479fb0aab2d607bd45b5708613c23cbc4af78ab0ef6bd, we will see that only block 804901 is That's right.

**Comment by Hosuke** [comment]
Ok, I may not be very familiar with bitcoin.
The dupes having different `block_height`:
https://dune.com/queries/2990773

**Comment by Longcnttbkhn** [comment]
@Hosuke  I have solved the duplicate error, please review it for me!

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @Longcnttbkhn 

[Sample transfers_bitcoin_satoshi](https://dune.com/queries/2992475)
[Sample balances_bitcoin_satoshi_day](https://dune.com/queries/2992482)

**Comment by longcnttbkhn** [comment]
[Demo dashboard](https://dune.com/longnhbkhn/bitcoin-balance-analyst)

**Comment by longcnttbkhn** [comment]
@jeff-dude @Hosuke Are there any more problems? I really need these tables for my work.

**Comment by jeff-dude** [comment]
> @jeff-dude @Hosuke Are there any problems? I really need these tables for my work.

i will look to do final review now, thank you for the patience!

**Comment by longcnttbkhn** [comment]
> i will look to do final review now, thank you for the patience!

Thank you, I'm honored to contribute!

**Review by jeff-dude — APPROVED** [review]
thanks again!

**Pull request merged**